*******************************************************************************
Date         : 2018-11-27
Author       : Jing Deng
Harman International Industries, Incorporated

Description  : This patch is for medialaunch connect MTP bugs. 
add a environment variable SMTPFS_DBPATH: config the database path of mtp stack, default /tmp/media/smtpfs.db
you can change the value eg. export SMTPFS_DBPATH=/opt/media/smtpfs.db
ChangeList   : 5304090
*******************************************************************************

diff --git a/src/simple-mtpfs-db.cpp b/src/simple-mtpfs-db.cpp
index 4a4bb95..e494675 100755
--- a/src/simple-mtpfs-db.cpp
+++ b/src/simple-mtpfs-db.cpp
@@ -71,38 +71,23 @@ int32_t SIMPLE_DB::doCommit(const string &sqlStr, DBQueryCallback cb, void *user
 SIMPLE_DB::SIMPLE_DB(const string DBName)
    : mPsqlite3(NULL)
 {
-   if(!DBName.empty())
+   string databaseName = DBName;
+   //get the SMTPFS_DB_PATH by environment 
+   char *dbPath = getenv(SMTPFS_DBPATH_ENV);
+
+   logmsg("SIMPLE_DB, env SMTPFS_DBPATH: ", dbPath, "\n");
+
+   if(dbPath != NULL)
    {
-      mPsqlite3 = doConnect(DBName);
+      databaseName = dbPath;
    }
+   
+   logmsg("SIMPLE_DB, real dbbase path: ", databaseName, "\n");
 
-#if 0
-   //create table
-   if(access(SMTPFSDBPATH, F_OK))
+   if(!databaseName.empty())
    {
-      string zSQLDevices = "CREATE TABLE mtpdevices (\
-                           uuid       TEXT PRIMARY KEY  DEFAULT '' NOT NULL,\
-                           mtime      INTEGER DEFAULT 0 NOT NULL\
-                           );";
-      
-      string zSQLFile = "CREATE TABLE mtpfile (\
-                           path       TEXT    DEFAULT '' NOT NULL,\
-                           mtpId      INTEGER DEFAULT 0 NOT NULL,\
-                           uuid       TEXT    DEFAULT '' NOT NULL\
-                           );";
-
-
-      if(doCommit(zSQLDevices, NULL, NULL, 0))
-      {
-         printf(" doCommit, create device table failed\n");
-      }
-
-      if(doCommit(zSQLFile, NULL, NULL, 0))
-      {
-         printf(" doCommit, create file table failed\n");
-      }
+      mPsqlite3 = doConnect(databaseName);
    }
-#endif
 }
 
 SIMPLE_DB::~SIMPLE_DB()
@@ -245,7 +230,7 @@ int32_t SIMPLE_DB::checkDeviceCount()
    //firset check device count, delete more MAX_MTPDEVICE_COUNT device
    int32_t devCount = 0;
    getDeviceCount(&devCount);
-   printf("writeDevice, devCount: %d\n", devCount);
+   //printf("writeDevice, devCount: %d\n", devCount);
    
    if(devCount >= MAX_MTPDEVICE_COUNT)
    {
diff --git a/src/simple-mtpfs-db.hpp b/src/simple-mtpfs-db.hpp
index 574be57..e75d9eb 100755
--- a/src/simple-mtpfs-db.hpp
+++ b/src/simple-mtpfs-db.hpp
@@ -27,7 +27,11 @@ typedef unsigned int uint32_t;
 typedef int int32_t;
 typedef int (*DBQueryCallback)(void*, int, char**, char**);
 
-#define SMTPFSDBPATH "/mnt/media/smtpfs.db"
+//default path of mtp stack db is /tmp/media/smtpfs.db
+//if you want to change it, you can config the environment SMTPFS_DB_PATH=xxxx
+#define SMTPFSDBPATH "/tmp/media/smtpfs.db"
+//smtpfs database env
+#define SMTPFS_DBPATH_ENV "SMTPFS_DBPATH"
 
 #define MAX_MTPDEVICE_COUNT (5)
